### PR TITLE
Backport of #4527 (Tables for alg param - QGIS Vector geometry -first part)

### PR DESCRIPTION
**Pull request by havatv**
Goal:

Ticket(s):  #4429. Backport of #4527.

- [ ] Backport to LTR documentation is required

### CODE REVIEW

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 1]
```diff
@@ -191,21 +320,59 @@ Polygon and line geometries are supported.
...
 
+.. seealso:: :ref:`qgisminimumboundinggeometry`
+
 Parameters
 ..........
 
 ``Input layer`` [vector: line, polygon]
   Input vector layer.
 
+Parameters
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: I do not recall whether I caught this in the master PR but we have two parameters sections for this alg

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 2]
```diff
@@ -226,86 +393,129 @@ case the buffer will result in a smaller polygon.
...
+       * 2 --- Bevel
+
+   * - **Miter limit**
+     - ``MITER_LIMIT``
+     - [number]
+
+       Default: 2.0
+     - Controls the maximum distance from the offset curve to use
+       when creating a mitered join (only applicable for miter
+       join styles.
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]:
```suggestion
       join styles).
```

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 3]
```diff
@@ -410,45 +711,146 @@ Outputs
...
+   * - Ring self-intersection 
+     - 
+     - 
+
+   * - Ring %1 of polygon %2 not in exterior ring
+     - 
+     - 
+
+   * - Polygon %1 lies inside polygon %2
+     - This error happens when a part of MultiPolygon geometry is inside in a hole of a MultiPolygon geometry.
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: Should this be 
```suggestion
     - This error happens when a part of MultiPolygon geometry is inside a hole of a MultiPolygon geometry. 
```
I guess it's if inside the hole of the same feature...

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 4]
```diff
@@ -469,26 +871,59 @@ See the 'Promote to multipart' or 'Aggregate' algorithms for alternative options
...
+   :header-rows: 1
+   :widths: 20 20 20 40
+   :stub-columns: 0
+
+   * - Label
+     - Name
+     - Type
+     - Description
+   * - **Collected**
+     - ``INPUT_LAYER``
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: As asked in the other PR...

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst (hunk 1 above)
Comment (havatv) [reply source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: Thanks . Will be fixed in the next commit.

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst (hunk 4 above)
Comment (havatv) [reply source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: Will fix.

**Pull request merged**